cg/apps/orderform/excel_orderform_parser.py

```diff
@@ -221,7 +213,8 @@ def parse_orderform(self, excel_path: str) -> None:
         if not raw_samples:
             raise OrderFormError("orderform doesn't contain any samples")
 
-        self.samples: List[ExcelSample] = parse_obj_as(List[ExcelSample], raw_samples)
+        excel_sample_list_validator = TypeAdapter(List[ExcelSample])
```

Comment:
From the docs, it seems like `TypeAdapter` is used to parse and validate objects which are not instantiated from a `BaseModel`. Is that right? So the use case is for when we have non-pydantic models that we want to validate?

https://docs.pydantic.dev/2.0/migration/#introduction-of-typeadapter

Reply:
I think it should be fine. They specifically list List[SomeModel] as a candidate to validate with TypeAdapter, which seems to be intended for all classes that need some validation:

"You may have types that are not BaseModels that you want to validate data against. Or you may want to validate a List[SomeModel], or dump it to JSON.

For use cases like this, Pydantic provides TypeAdapter, which can be used for type validation, serialization, and JSON schema generation without creating a BaseModel."

https://docs.pydantic.dev/latest/usage/type_adapter/